core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -314,30 +319,46 @@ private boolean isIntermediateTopic(final String topic) {
         return options.valuesOf(intermediateTopicsOption).contains(topic);
     }
 
-    private void maybeDeleteInternalTopics(final ZkUtils zkUtils) {
+    private void maybeDeleteInternalTopics(final KafkaAdminClient adminClient) {
 
         System.out.println("Deleting all internal/auto-created topics for application " + options.valueOf(applicationIdOption));
-
-        for (final String topic : allTopics) {
-            if (isInternalTopic(topic)) {
-                try {
-                    if (!dryRun) {
-                        final TopicCommand.TopicCommandOptions commandOptions = new TopicCommand.TopicCommandOptions(new String[]{
-                            "--zookeeper", options.valueOf(zookeeperOption),
-                            "--delete", "--topic", topic});
-                        TopicCommand.deleteTopic(zkUtils, commandOptions);
-                    } else {
-                        System.out.println("Topic: " + topic);
-                    }
-                } catch (final RuntimeException e) {
-                    System.err.println("ERROR: Deleting topic " + topic + " failed.");
-                    throw e;
+        List<String> topicsToDelete = new ArrayList<>();
+        for (final String listing : allTopics) {
+            if (isInternalTopic(listing)) {
+                if (!dryRun) {
+                    topicsToDelete.add(listing);
+                } else {
+                    System.out.println("Topic: " + listing);
                 }
             }
         }
+        if (!dryRun) {
+            doDelete(topicsToDelete, adminClient);
+        }
         System.out.println("Done.");
     }
 
+    private void doDelete(List<String> topicsToDelete, KafkaAdminClient adminClient) {
+        RuntimeException deleteException = null;
+        DeleteTopicsResult deleteTopicsResult = adminClient.deleteTopics(topicsToDelete);
```

Comment:
Nit: `final` -- also some more below

Reply:
Same as above: does `deleteTopics` throw any exceptions? If yes should we handle it?

Reply:
@guozhangwang yes,  but not directly.

Any exception occurring while deleting the topic is thrown with the `KafkaFuture.get` call for the given topic name from the 'Map<String, KafkaFuture>` map on line 345.

We check for exceptions by iterating over the map on line 347, and any exceptions are logged out, and the first exception is captured and is propagated up to the `run` method which handles the exception on line 118.